Kernel/Devices/OffsetDiskDevice.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <AK/RetainPtr.h>
+#include <Kernel/Devices/DiskDevice.h>
+
+class OffsetDiskDevice final : public DiskDevice {
+public:
+    static Retained<OffsetDiskDevice> create(Retained<DiskDevice>&& device, unsigned offset);
```

Comment:
Longer-term I think this should be a `DiskPartition` class that has an offset but also a `length`.
We could then have `DiskDevice` read the partition table and instantiate a number of `DiskPartition` objects.

Also maybe we should call the `offset` argument here `block_offset` just to make things clear.

Reply:
I did wonder about the naming for this thing. I followed the example from `FileBackedDiskDevice`, but I'm not picky about what it's called.

Reply:
Then let's call it `DiskPartition` right away and over time we can massage it into being more of a fully-fledged partition object. :)

Reply:
Done and done!